webfrontend/src/components/setup/setup-tags.tsx

```diff
@@ -0,0 +1,342 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { triggerErrorMessage } from "../../utils/locals";
+import { Tag } from "../posts/one-post";
+import { postRequestContent } from "../../utils/request";
+import { Button } from "react-bootstrap";
+import Modal from "react-bootstrap/Modal";
+
+
+const SetupTags = () => {
+  const [tags, setTags] = useState<Array<Tag>>([]);
+  const [searchParam, setSearchParam] = useState("");
+  const [searchResults, setSearchResults] = useState<Array<Tag>>([]);
+  const [toBeSubmitted, setToBeSubmitted] = useState<Array<Tag>>([]);
+  const [tagCountLimit, setTagCountLimit] = useState(0);
+  const searchTagForm = useRef<HTMLFormElement>(null);
+  const [showTagResult, setShowTagResult] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+  const [searchDone, setSearchDone] = useState(false);
+  const [canRemoveTag, setCanRemoveTag] = useState(false);
+  const [showRemoveAlert, setShowRemoveAlert] = useState(false);
+  const [tagToBeRemoved, setTagToBeRemoved] = useState<Tag>();
+  const [addTagMessageButton, setAddTagMessageButton] =
+    useState<boolean>(false);
```

Comment:
booleans are better starting with `is`, so that anywhere in the code that references this variable knows that this is a boolean.